docs/design/crd-validation/README.md

```diff
@@ -0,0 +1,79 @@
+
+As shown in the above diagram, the admission control process proceeds in two phases. In the first phase, mutating admission controllers are run. In the second phase, validating admission controllers are run. Note again that some of the controllers are both.
+
+The admission controllers intercept requests based on the values given in the configuration. In this config, we have to provide the details on 
+* What resources should it be looking for ? (Pods, Service)
+* What api version and group does it belong to ? *Example : ApiVersion = (v1, v1beta) Group version = (admissionregistration.k8s.io)*
+* What kind of operation should it intercept ? (Create, Update, Delete)
+* A valid base64 encoded CA bundle.
+* What path do we want to send with HTTPs request (/validate, /mutate)
+
+A webhook server should be in place (*with valid TLS certificates*) to intercept any HTTPs request that comes with the above path value. Once the request is intercepted by the server, an [AdmissionRequest](https://github.com/kubernetes/api/blob/master/admission/v1beta1/types.go#L40) object is sent through with the resource specifications. 
+
+When the webhook server receives Admission Request, it will perform predefined validations on the provided resource values and send back an [AdmissionResponse](https://github.com/kubernetes/api/blob/master/admission/v1beta1/types.go#L116) with the indication whether request is accepted or rejected.
+
+If any of the controllers in either phase reject the request, the entire request is rejected immediately and an error is returned to the end-user.
+
+## Certificate Management in OCP
+
+![Prod_AC_Image](./ac-ocp.png)
+
+As shown in the above diagram, In the OCP cluster, The management of certificates would be completely done by the Openshift Service CA Operator. This operator would be responsible for handling all the certificate management duties like generating and renewing them when OCS is deployed in Openshift Environment.
+
+The first step would be to generate the certificates and assign them to the proper components. On annotating any service with 
+
+`service.beta.openshift.io/serving-cert-secret-name:<secret-name>` 
+
+We have to deploy the the annotated service at the start to generate the signed certificates. This can done manually  through the command 
+
+`oc create -f ${SERVICE_NAME}.yaml`
+
+For verifying 
+
+`oc describe ${SERVICE_NAME} -n ${NAMESPACE}`
+
+**Output**
+```
+oc describe service <service-name>
+...
+Annotations:              service.beta.openshift.io/serving-cert-secret-name: <service-name>
+                          service.beta.openshift.io/serving-cert-signed-by: openshift-service-serving-signer@1556850837
+```
+The Openshift Sevice CA operator automatically generates a signed serving certificate and key pair into a secret in the same namespace as the service. 
+This secret will be mounted onto the server pod. Our server would be looking for tls certificates and automatically set them for TLS communication.
```

Comment:
We'll need to understand the sequence of events. For example, would the OCS operator wait for the secret to be generated before launching the admission controller server? Or could we rely on the pod startup to wait for the secret to exist before starting? The latter seems better, but the pod would also appear in a failed state for some (hopefully short) time until the secret is created.

Reply:
added a section for `setup in ocs` which addresses this